### PR TITLE
Move glut dependency from dart-core to dart

### LINKED ISSUE #329: OSX build problem related to GLUT
A user had trouble building gazebo ( https://github.com/osrf/homebrew-simulation/issues/36 ) due to a linking problem with GLUT when not building with `BUILD_CORE_ONLY=ON`. Are the GLUT libraries being exported as dependencies in the cmake config or pkg-config files? If they are not needed by downsteam users, we should just link the GLUT libraries with the apps and not the dart library.

As an aside, GLUT is only used by the `dart/gui` and `apps` code, which should only be built when `BUILD_CORE_ONLY=ON`. However, I see that [GLUT is always searched for](https://github.com/dartsim/dart/blob/master/CMakeLists.txt#L128-L140) (even when not `CORE_ONLY`), and it's library and header files are also always used.

### PR DESCRIPTION
**Pull request by jslee02**
Fixes #329.

### CODE REVIEW

File: dart/gui/LoadGlut.h [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+ *   POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#ifndef DART_GUI_LOADGLUT_H_
+#define DART_GUI_LOADGLUT_H_
+
+#if WIN32
+  #include <cstdlib> // To disable glut::exit() function
+  #include <GL/glut.h>
+#elif defined(__linux)
```
Comment (scpeters) [inline dart/gui/LoadGlut.h]: `__linux__` is the [POSIX compliant macro](http://sourceforge.net/p/predef/wiki/OperatingSystems/) as @j-rivero pointed out in https://github.com/simbody/simbody/pull/338

File: dart/renderer/LoadOpengl.h [hunk 2]
```diff
@@ -34,19 +34,21 @@
...
+#ifndef DART_RENDERER_LOADOPENGL_H_
+#define DART_RENDERER_LOADOPENGL_H_
 
 #if WIN32
-#include <cstdlib> // To disable glut::exit() function
-#include <GL/glut.h>
+  #include <windows.h>
+  #include <GL/gl.h>
+  #include <GL/glu.h>
 #elif defined(__linux)
```
Comment (scpeters) [inline dart/renderer/LoadOpengl.h]: `__linux__` here too

File: dart/gui/LoadGlut.h (hunk 1 above)
Comment (jslee02) [inline dart/gui/LoadGlut.h]: It is replaced by __linux__, thanks!

**Comment by scpeters** [comment]
+1 pending CI result

**Comment by scpeters** [comment]
thanks!

**Pull request merged**